lighthouse-plugin-publisher-ads/utils/resource-classification.js

```diff
@@ -37,12 +37,67 @@ function isGoogleAds(url) {
 }
 
 /**
- * Checks if the url is for pubads implementation tag.
+ * Checks if the url is loading an AdSense script.
  * @param {URL|string} url
  * @return {boolean}
  */
-function isImplTag(url) {
-  return /(^\/gpt\/pubads_impl_\d+.js)/.test(toURL(url).pathname);
+function isAdSenseTag(url) {
+  const {host, pathname} = toURL(url);
+  const matchesHost = [
+    'pagead2.googlesyndication.com'].includes(host);
+  const matchesPath =
+      [
+        '/pagead/js/adsbygoogle.js',
+        '/pagead/show_ads.js',
+      ].includes(pathname);
+  return matchesHost && matchesPath;
+}
+
+/**
+ * Checks if the url is loading an AdSense script.
+ * @param {URL|string} url
+ * @return {boolean}
+ */
+function isAdSenseImplTag(url) {
+  const {host, pathname} = toURL(url);
+  const matchesHost = [
+    'pagead2.googlesyndication.com'].includes(host);
+  const matchesPath =
+      /(^\/pagead\/js\/.*\/show_ads_impl\.js)/.test(pathname);
+  return matchesHost && matchesPath;
+}
+
+/**
+ * Checks if the url is loading an AdSense loader or impl script.
+ * @param {URL} url
+ * @return {boolean}
+ */
+function isAdSense(url) {
+  return isAdSenseTag(url) || isAdSenseImplTag(url);
+}
+
+/**
+ * Checks if a network request is an AdSense ad request.
+ * @param {LH.Artifacts.NetworkRequest} request
+ * @return {boolean}
+ */
+function isAdSenseAdRequest(request) {
+  if (!request) return false;
+  const url = new URL(request.url);
+  return (
+    url.pathname === '/pagead/ads' &&
+    url.host === 'googleads.g.doubleclick.net'
+  );
+}
+
+/**
+ * @param {Artifacts['IFrameElement']} iframe
+ * @return {boolean}
+ */
+function isAdSenseIframe(iframe) {
+  // TODO(jonkeller) Discuss with warrengm@ which of these is correct
+  // return /(^google_ads_frame)/.test(iframe.id);
+  return /(^aswift_\d+)/.test(iframe.id);
```

Comment:
@warrengm Let's talk sometime today?
Basically the issue is that it's rendered like <iframe id="aswift_\d+">...<iframe id="google_ads_frame\d+" src="somewhere-cross-domain">
In both cases (the line that's commented out here, and the line that isn't) we end up here: https://github.com/googleads/publisher-ads-lighthouse-plugin/blob/e57628a1fb0787bd5eb3d9a004325e50425077b6/lighthouse-plugin-publisher-ads/audits/first-ad-paint.js#L74

Reply:
Talked/fixed.